jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py

```diff
@@ -58,25 +66,19 @@ def encode(
             scalar value.
         :param parameters: dictionary to defines the `traversal_paths`.
         """
+        if not docs:
+            return
 
-        if docs:
-            cleaned_document_array = self._get_input_data(docs, parameters)
-            self._create_embeddings(cleaned_document_array)
-
-    def _get_input_data(self, docs: DocumentArray, parameters: dict):
-        """Create a filtered set of Documents to iterate over."""
-
-        traversal_paths = parameters.get('traversal_paths', self.traversal_paths)
-
-        # traverse thought all documents which have to be processed
-        flat_docs = docs.traverse_flat(traversal_paths)
-
-        # filter out documents without audio wav
-        filtered_docs = DocumentArray(
-            [doc for doc in flat_docs if doc.blob is not None]
+        batch_generator = get_docs_batch_generator(
+            docs,
+            traversal_path=parameters.get('traversal_paths', self.traversal_paths),
+            batch_size=parameters.get('batch_size', self.batch_size),
+            needs_attr='blob',
         )
 
-        return filtered_docs
+        with torch.no_grad():
+            for batch in batch_generator:
```

Comment:
```suggestion
            for batch in docs.batch(batch_size, traversal_paths, required_attr='blob'):
```

Reply:
ah i didn't even know we have batch already , great!